### ISSUE TITLE
install-package get the same version

### ISSUE THREAD

**Issue by loonghao** [mottosso/rez-pipz#37]
Hi @mottosso 

I via command `rez env pipz -- install rez_tools` will get the same version from pip install.

[rez_tools](https://github.com/loonghao/rez-tools)

```
Using python-3.7
Using pip-20.2b1
Using pipz-1.3.0
Reading package lists... ok - 3.45s
Discovering existing packages... 7.1.2 click
{'pipz': True, 'version': '7.1.2', 'name': 'click', 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"}
15:39:17 INFO     Applying global preprocess function
INFO Applying global preprocess function
{'pipz': True, 'version': '7.1.2', 'name': 'click', 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"}
7.1.2 click test
click 7.1.2
5.3.1 PyYAML
{'pipz': True, 'version': '5.3.1', 'name': 'PyYAML', 'variants': [['os-windows-10', 'python-3.7']], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"}
15:39:17 INFO     Applying global preprocess function
INFO Applying global preprocess function
{'pipz': True, 'version': '7.1.2', 'name': 'PyYAML', 'variants': [['os-windows-10', 'python-3.7']], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"} sd
15:39:17 INFO     Package attributes were changed in preprocessing:
Added attributes: ['package_cls', 'installed_variants', 'skipped_variants']
Changed attributes: ['version']
INFO Package attributes were changed in preprocessing:
Added attributes: ['package_cls', 'installed_variants', 'skipped_variants']
Changed attributes: ['version']
{'pipz': True, 'version': '7.1.2', 'name': 'PyYAML', 'variants': [['os-windows-10', 'python-3.7']], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')", 'package_cls': <class 'rez.developer_package.DeveloperPackage'>, 'installed_variants': [], 'skipped_variants': []}
7.1.2 PyYAML test
PyYAML 7.1.2
0.2.3 rez-tools
{'pipz': True, 'version': '0.2.3', 'name': 'rez_tools', 'requires': ['click>=7.0', 'pyyaml>=5.0'], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"}
15:39:17 INFO     Applying global preprocess function
INFO Applying global preprocess function
{'pipz': True, 'version': '7.1.2', 'name': 'rez_tools', 'requires': ['click>=7.0', 'pyyaml>=5.0'], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')"} sd
15:39:17 INFO     Package attributes were changed in preprocessing:
Added attributes: ['package_cls', 'installed_variants', 'skipped_variants']
Changed attributes: ['version']
INFO Package attributes were changed in preprocessing:
Added attributes: ['package_cls', 'installed_variants', 'skipped_variants']
Changed attributes: ['version']
{'pipz': True, 'version': '7.1.2', 'name': 'rez_tools', 'requires': ['click>=7.0', 'pyyaml>=5.0'], 'commands': "env.PATH.prepend('{root}/bin')\nenv.PYTHONPATH.prepend('{root}/python')", 'package_cls': <class 'rez.developer_package.DeveloperPackage'>, 'installed_variants': [], 'skipped_variants': []}
7.1.2 rez_tools test
rez_tools 7.1.2
ok - 0.07s
The following NEW packages will be installed:
  click        7.1.2
  PyYAML       7.1.2  os-windows-10/python-3.7
  rez_tools    7.1.2
```
If I disable the following lines, there will be no such issues.

https://github.com/mottosso/rez-pipz/blob/152307c63ede54ca9d71e988b53cb1200650919b/python/pipz/pip.py#L240-L243

let me know if you have any idea about that.

**Comment by davidlatwe** [comment]
```
F:\>rez-env pipz -- install rez_tools --prefix c:/test/pipz
Using python-3.7
Using pip-20.2b1
Using pipz-1.3.0
...
The following NEW packages will be installed:
  click        7.1.2  default
  PyYAML       5.3.1  os-windows-10.0/python-3.7
  rez_tools    0.2.3  default
```

Just had a quick test and looks I have no such issue. (dependency package's version didn't get overrided)

> If I disable the following lines, there will be no such issues.

That part of code is to appliy [`rezconfig.package_preprocess_function`](https://github.com/nerdvegas/rez/wiki/Configuring-Rez#package_preprocess_function), implemented in #31.

Could it be your package preprocess function changed the pacakge version ?

**Comment by loonghao** [comment]
@davidlatwe Thanks. yes. due to I have custom package_preprocess_function. closing the issues.
